test-jest/chart/AreaChart.spec.tsx

```diff
@@ -222,5 +220,38 @@ describe('AreaChart', () => {
       spies.forEach(el => expect(el).toHaveBeenCalledTimes(1));
       expect(axisSpy).toHaveBeenCalledTimes(2);
     });
+
+    test('should only show the last data when the brush travelers all moved to the right', () => {
+      const { container } = render(chart);
+
+      const leftBrushTraveler = container.querySelector('.recharts-brush-traveller');
+
+      const mouseDownEvent = mockMouseEvent('mousedown', leftBrushTraveler!, { pageX: 0, pageY: 0 });
+      const mouseMoveEvent = mockMouseEvent('mousemove', window, { pageX: 400, pageY: 0 });
+
+      mouseDownEvent.fire();
+      mouseMoveEvent.fire();
+      fireEvent.mouseUp(window);
+
+      expect(leftBrushTraveler?.firstChild).toHaveAttribute('x', '390'); // not sure why the drag end at x: 390
+      expect(container.querySelectorAll('.recharts-area-dot')).toHaveLength(1);
+    });
+
+    test('should only show the first data when the brush travelers all moved to the left', () => {
+      const { container } = render(chart);
+
+      const rightBrushTraveler = container.querySelectorAll('.recharts-brush-traveller')[1];
+
+      const mouseDownEvent = mockMouseEvent('mousedown', rightBrushTraveler!, { pageX: 400, pageY: 0 });
+      const mouseMoveEvent = mockMouseEvent('mousemove', window, { pageX: 0, pageY: 0 });
+
+      mouseDownEvent.fire();
+      mouseMoveEvent.fire();
+      fireEvent.mouseUp(window);
+
+      // not sure why the drag ended at x: 65, but it did close to the 1st point but not 2nd
```

Comment:
just for my understanding this (and the above test) is saying that when brush is moved all the way to the left or the right the attributes stop being calculated after a certain point? Maybe once only one point is present it stops caring?

Would be nice to understand this (not in context of this test but just in general)